### PR TITLE
[LoRA][II] Add fused MOE LoRA Triton kernel and tests

### PR DESCRIPTION
**Pull request by yushengsu-thu**
Split this PR https://github.com/sgl-project/sglang/pull/14105 into 3 parts - Part II

Add Triton-based fused MoE LoRA kernel for combined expert routing and LoRA computation:
- fused_moe_lora_kernel.py: Triton kernels for fused_moe_lora_shrink and fused_moe_lora_expand
- test_fused_moe_lora_kernel.py: Unit tests validating kernel correctness against PyTorch reference
- Update triton_ops/__init__.py to export fused_moe_lora

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/lora/triton_ops/fused_moe_lora_kernel.py [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+    if (ptr_tensor := _LORA_PTR_DICT.get(key)) is not None:
+        return ptr_tensor
+
+    tensor_ptrs = []
+    for lora_weight in lora_weights:
+        tensor_ptrs.append(lora_weight.data_ptr())
+    ptr_tensor = torch.tensor(tensor_ptrs, device=device, dtype=torch.uint64)
+
+    _LORA_PTR_DICT[key] = ptr_tensor
+    return _LORA_PTR_DICT.get(key)
```
[the remark this answers is not in the document]
Comment (sshleifer) [reply python/sglang/srt/lora/triton_ops/fused_moe_lora_kernel.py:37]: Is there logic to skip loras that are not currently being used?

**Review by sshleifer — APPROVED** [review]
LGTM, we can try to speed it up in followup

**Review by Fridge003 — APPROVED** [review]
